### ISSUE TITLE
Refund transaction link point to transaction that does not exist in blockchain

### ISSUE THREAD

**Issue by thomaseizinger** [itchysats/itchysats#800]
![image](https://user-images.githubusercontent.com/5486389/144765476-084ae041-7d5a-4d05-9160-182d355ca773.png)

This is a weird one. Here are the logs:

```
2021-12-05 21:52:22 DEBUG daemon::cfd_actors: Learnt latest oracle attestation for event: /x/BitMEX/BXBT/2021-12-02T23:00:00.price?n=20
2021-12-05 21:52:22 ERROR daemon::cfd_actors: Error when trying to publish CET: Cannot publish CET in state Pending Open
2021-12-05 21:52:22  INFO daemon::oracle: Fetched new attestation for /x/BitMEX/BXBT/2021-12-02T23:00:00.price?n=20
2021-12-05 21:52:22 DEBUG daemon::cfd_actors: Learnt latest oracle attestation for event: /x/BitMEX/BXBT/2021-12-02T23:00:00.price?n=20
2021-12-05 21:52:22 ERROR daemon::cfd_actors: Error when trying to publish CET: Cannot publish CET in state Pending Open
2021-12-05 21:52:31 DEBUG daemon::monitor: Found relevant Bitcoin transaction txid=b887333aee59c89f249d81c8486a1c212917452b7a8d3f84c0d6a5985d6df63d status=confirmed with 367 blocks
2021-12-05 21:52:31 DEBUG daemon::monitor: Found relevant Bitcoin transaction txid=0de71f7ad590b930887a872ccdf58e290328c94d4ab560c3d239e1bcc819204a status=confirmed with 274 blocks
2021-12-05 21:52:31 DEBUG daemon::monitor: Found relevant Bitcoin transaction txid=11716cbabda362a967a18f49ac6ed479ced384ce19d303e1f21ec588819d36c9 status=confirmed with 262 blocks
2021-12-05 21:52:31  INFO daemon::monitor: Bitcoin transaction reached monitoring target txid=b887333aee59c89f249d81c8486a1c212917452b7a8d3f84c0d6a5985d6df63d target=confirmed with 1 blocks
 current=confirmed with 367 blocks
2021-12-05 21:52:31  INFO daemon::monitor: Bitcoin transaction reached monitoring target txid=0de71f7ad590b930887a872ccdf58e290328c94d4ab560c3d239e1bcc819204a target=confirmed with 1 blocks
 current=confirmed with 274 blocks
2021-12-05 21:52:31 DEBUG daemon::model::cfd: Was in unexpected state Open, jumping ahead to OpenCommitted order_id=d728add5-b39e-42a5-9545-a5e05474f0cf
2021-12-05 21:52:31 DEBUG daemon::cfd_actors: The cfd is not ready for CET publication yet: Unprepared
2021-12-05 21:52:31  INFO daemon::monitor: Bitcoin transaction reached monitoring target txid=0de71f7ad590b930887a872ccdf58e290328c94d4ab560c3d239e1bcc819204a target=confirmed with 12 block
s current=confirmed with 274 blocks
2021-12-05 21:52:32 DEBUG daemon::cfd_actors: The cfd is not ready for CET publication yet: TimelockExpired
2021-12-05 21:52:32  INFO daemon::monitor: Bitcoin transaction reached monitoring target txid=0de71f7ad590b930887a872ccdf58e290328c94d4ab560c3d239e1bcc819204a target=confirmed with 216 bloc
ks current=confirmed with 274 blocks
2021-12-05 21:52:32 ERROR daemon::taker_cfd: Message handler failed: Failed to publish CET: Broadcasting transaction failed. Txid: 64c837c9589879e2f1ff75dd332c6a404c20405ccd0c600a540fa649a1
4faf35. Raw transaction: 020000000001014a2019c8bce139d2c360b54a4dc92803298ef5cd2c877a8830b990d57a1fe70d0000000000d800000002ce5105000000000016001439e9ace111e1c5db3670a96589a7f7e0d6efd9b58ea8
020000000000160014d519e33f57fe46a8d07b7393b5b7f91a956ddbc4034730440220390cbca1c63da97de9516ef4024479c64f0a8bb2ea94d6d829d31f14c297921302203820abcc3630b0a084f96ee55cf2307614835dce392bb8829f8
90461765150d901473044022048099dd6b20a1a29db53470c13cd306b3314eeb7c73aa7a45b795dcc3d81706a022029f6723919d4586aa9d38981389bdb556af4faa54e682f15fd9bf31a775c16ef01e021033af2662138adeca67440ec71
4db9f9753bb967378220a905c609d9e79e1a91cdac646376a914ad06016bbba917bed3239c372df3f20ffc9f81c788ad76a914a4c9b4430655b7094b42f73f6537f914a6c4650288ad76a914053ac52d5f7da91386d17a2b4af3555efb9ef
e92886776a914b2ea7b280bef18832e28b351a0dc782f27b2402888ad76a914a81e4b3b3db27e25e47cd25a6239f419201aa64e88ad76a914661728af5b52bd5907178bcdc3bcf6e7e4798c8d8868672102d622746d5456158dfbe7f5fe0a
ce8cb7fd30ff34163b6bc0bcf097ad36e9d46668ac00000000: Electrum(Protocol(String("sendrawtransaction RPC error: {\"code\":-25,\"message\":\"bad-txns-inputs-missingorspent\"}")))
2021-12-05 21:52:32  INFO daemon::monitor: Bitcoin transaction reached monitoring target txid=11716cbabda362a967a18f49ac6ed479ced384ce19d303e1f21ec588819d36c9 target=confirmed with 1 blocks
 current=confirmed with 262 blocks
2021-12-05 21:52:32 ERROR daemon::taker_cfd: Message handler failed: No attestation available when reaching CET finality
2021-12-05 21:52:32  INFO daemon::monitor: Bitcoin transaction reached monitoring target txid=11716cbabda362a967a18f49ac6ed479ced384ce19d303e1f21ec588819d36c9 target=confirmed with 1 blocks
 current=confirmed with 262 blocks
2021-12-05 21:52:32 ERROR daemon::taker_cfd: Message handler failed: No attestation available when reaching CET finality
```

As we can see, we tried to publish a CET with txid: `64c837c9589879e2f1ff75dd332c6a404c20405ccd0c600a540fa649a14faf35` but it failed with `bad-txns-inputs-missingorspent`. This is the commit transaction: https://mempool.space/testnet/tx/11716cbabda362a967a18f49ac6ed479ced384ce19d303e1f21ec588819d36c9. The output of `0.00170704` BTC is suspiciously similar to my initial margin of `0.001744` (minus a couple of fees perhaps?) so maybe the CFD got refunded.

The link in the UI points to here: https://mempool.space/testnet/tx/64c837c9589879e2f1ff75dd332c6a404c20405ccd0c600a540fa649a14faf35

Which is the same TX-ID as the CET that we tried to publish.

**Comment by da-kami** [comment]
We should not transition **after** we are sure the transaction was included! Investigation needed why there was a transition into `Refunded` if the transaction was not included.

**Comment by da-kami** [comment]
Closed in favour of #873 
Additionally this should be more resilient/solved once we have the event sourcing model.
